src/Service/PubliccodeService.php

```diff
@@ -0,0 +1,836 @@
+     *
+     * GITLAB: The logo can be given in multiple ways for gitlab. (what we have seen)
+     * 5. An url to the logo. Here we don't validate the gravatar url. TODO: validate the given gravatar url. (https://www.gravatar.com/avatar/e64c7d89f26bd1972efa854d13d7dd61)
+     * 6. A upload gitlab url of the logo. (https://gitlab.com/uploads/-/system/project/avatar/33855802/760205.png)
+     * 7. A github url to where the logo is placed in a repository. (https://gitlab.com/discipl/RON/regels.overheid.nl/-/blob/master/images/WORK_PACKAGE_ISSUE.png)
+     * 8. A relative path. From the root of the repository to the image. (/images/WORK_PACKAGE_ISSUE.png)
+     *
+     * @param array        $publiccodeArray The mapped publiccode array from the github api.
+     * @param Source       $source          The github api source.
+     * @param ObjectEntity $url             The url of the repository or organization object.
+     *
+     * @return string|null The logo from the publiccode
+     */
+    public function handleLogo(array $componentArray, Source $source, string $url, string $repositoryId): ?string
+    {
+        // Parse url to get the path (organization and repository) from the repository url.
+        // The repositoryName is used for option 2, 3 and 4.
+        $repositoryName = \Safe\parse_url($url)['path'];
+
+        // The logo can be given in multiple ways. (what we have seen). Check the function tekst for explanation about the types we handle.
+        // Check if the logo is a valid url.
+        if (filter_var($componentArray['logo'], FILTER_VALIDATE_URL) !== false) {
+            $this->pluginLogger->info('The logo is a valid url. Check whether the logo comes from source https://avatars.githubusercontent.com, https://www.gravatar.com or whether the logo must be retrieved from the github or gitlab api with the given logo URL.');
+
+            // Parse url to get the host and path of the logo url.
+            $parsedLogo = \Safe\parse_url($componentArray['logo']);
+
+            // There should always be a host because we checked if it is a valid url.
+            $domain = $parsedLogo['host'];
+            switch ($domain) {
+                // Check if the logo is as option 1, a logo from https://avatars.githubusercontent.com.
+                // Check if the domain is https://avatars.githubusercontent.com. If so we don't have to do anything and return the publiccodeArray logo.
+            case 'avatars.githubusercontent.com':
+                // TODO: Validate the avatar url. Call the source with path and check is the status code is 200. The function handleRawLogo can be used for this.
+                $this->pluginLogger->info('The logo from the publiccode file is from https://avatars.githubusercontent.com. Do nothing and return the url.');
+
+                // Return the given avatar logo url.
+                return $componentArray['logo'];
+                    break;
```

Comment:
You don't need to break if you already return something (same for other cases below)